zeek_log_transport.sh

```diff
@@ -237,12 +279,23 @@ status "Sending logs to rita/aihunter server $aih_location , My name: $my_id , l
 status "Preparing remote directories"
 ssh $extra_ssh_params "$aih_location" "mkdir -p ${remote_top_dir}/$today/ ${remote_top_dir}/$yesterday/ ${remote_top_dir}/$twoda/ ${remote_top_dir}/$threeda/ ${remote_top_dir}/current/"
 
+recv_loc="`pwd`/zeek-log-transport.yaml"
+request_logs=`get_send_logs $aih_location "/etc/AI-Hunter/zeek-log-transport.yaml" "$recv_loc"`
+request_days=`get_send_days $recv_loc`
+logs_str=`echo "${request_logs[*]// /|}"`
+
 cd "$local_tld" || fail "Unable to change to $local_tld"
-send_candidates=`find . -type f -mtime -3 -iname '*.gz' | egrep '(conn|dns|http|ssl|x509|known_certs)' | sort -u`
+
+query="find . -type f -mtime -$request_days -iname '*.gz' | egrep '($logs_str)' | sort -u"
+send_candidates=$(eval "$query")
```

Comment:
I think this will evaluate to a string, and I think it should be quoted: `"$(eval "$query")"`

Reply:
Testing this works with quotes around the whole command `"$(eval $query)"` or around query like so `$(eval "$query")` but I can't have quotes around both like `"$(eval $"query")"`
I will push up a change for the quotes around the whole command

Reply:
You can have quotes like `"$(eval "$query")"` (but not like you it with `$"query"`). From other languages, you'd think that the nested quotes would conflict with each other but `$()` is a subshell and my understanding is any quotes inside are not matched with anything outside it.

Here's an example:
```bash
$ cat test.sh
#!/bin/bash

print_num_args() {
    echo $#
}

args="one two"

# this treats $args as two separate arguments
how_many="$(print_num_args $args)"
echo "without inner quotes: $how_many"

# this treats $args as a single argument
how_many="$(print_num_args "$args")"
echo "with inner quotes: $how_many"

```

```bash
$ bash test.sh
without inner quotes: 2
with inner quotes: 1
```

So the difference is how bash treats the argument when it has whitespace in it. Bash will split unquoted arguments on whitespace and quoted arguments will not be split.

That being said, I think you're ok in this case. I read the help page and `eval` takes multiple arguments and just concatenates them together with a space before running.

And I also just learned something. The quotes _outside_ the subshell appear to be unnecessary even with whitespace. Take a look at this example:

```bash
$ cat test.sh       
#!/bin/bash

test1=one two three
echo "$test1"

test2=$(echo one two three)
echo "$test2"
```

The only difference in setting these two variables is that one is in a subshell and one is not. They are both unquoted. But the one in the subshell works and the other doesn't.

```bash
$ bash test.sh      
test.sh: line 3: two: command not found

one two three
```

So with the behavior of eval and subshells combined, I think in this case quotes are completely unnecessary for this particular case. All of these should work and have slightly different nuances. And now I prefer your original line of code lol.
- `$(eval $query)`
- `$(eval "$query")` - This will likely be my pick in my code going forward since it reduces the quote soup. Until I learn about some new nuance of bash that bites me in the future.
- `"$(eval $query)"`
- `"$(eval "$query")"`

Bash is weird :man_facepalming: